unit/java_bytecode/java_object_factory/gen_nondet_string_init.cpp

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************\
+
+ Module: Java string library preprocess.
+         Test for converting an expression to a string expression.
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <util/expr.h>
+#include <util/std_code.h>
+#include <util/namespace.h>
+#include <java_bytecode/java_object_factory.h>
+#include <java_bytecode/java_bytecode_language.h>
+#include <java_bytecode/java_root_class.h>
+#include <langapi/language_util.h>
+#include <langapi/mode.h>
+#include <iostream>
+
+TEST_CASE("Generate string object")
+{
+  source_locationt loc;
+  symbol_tablet symbol_table;
+  register_language(new_java_bytecode_language);
+
+  // Add java.lang.Object to symbol table
+  symbolt jlo_sym;
+  jlo_sym.name="java::java.lang.Object";
+  jlo_sym.type=struct_typet();
+  jlo_sym.is_type=true;
+  java_root_class(jlo_sym);
+  bool failed=symbol_table.add(jlo_sym);
+  CHECK_RETURN(!failed);
+
+
+  // Add java.lang.Object
+  symbolt new_symbol;
```

Comment:
You appear to make two almost identical symbols for the java.lang.Object here. Is this intentional

Consider pulling it out into a helper function at the top to reduce boiler plate code in the test itself.